core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -90,6 +90,25 @@ static boolean hasIds(Schema schema) {
     return AvroCustomOrderSchemaVisitor.visit(schema, new HasIds());
   }
 
+  /**
+   * Check if any of the nodes in a given avro schema is missing an ID recognizable by Iceberg core API
```

Comment:
This is the Iceberg core API, so it doesn't make sense to refer to it as something else. You can just say "Check if the schema is missing any ID properties." And below, you can clarify which IDs are considered.